x-pack/solutions/security/plugins/security_solution/server/lib/detection_engine/rule_types/indicator_match/threat_mapping/create_event_signal.ts

```diff
@@ -109,7 +110,18 @@ export const createEventSignal = async ({
       }
     }
 
-    const ids = Array.from(signalsQueryMap.keys());
+    const { validEvents, invalidIds } = validateCompleteThreatMatches(
+      signalsQueryMap,
+      threatMapping
+    );
+
+    if (invalidIds.length > 0) {
+      ruleExecutionLogger.debug(
+        `Invalid signals: ${invalidIds.join(', ')} - skipping these signals`
```

Comment:
maybe not invalid, but not matched? Invalid may imply documents are corrupted or not processed properly, where in facts they are not matching search criteria